### PR TITLE
fix: Popup within Popup clicking

### LINKED ISSUE #7970: The property filter pop-up closes if you change the selected property
## Bug description

When you change the selected property in the property filter (as opposed to originally selecting it) the filter pop-up closes.

property-change

## Expected behavior

When first choosing the property. The operator selector focuses when the property is changed. So the behavior should be the same when changing the property

## How to reproduce

1. visit any page with an "add filter" button
2. click add filter
3. choose a property, operator, and value
4. click on the blue property pill
5. observe the property opens
5. change the property
6. observe the pop up closes

## Environment

- [x] PostHog Cloud

### LINKED ISSUE #11814: filter selector closes before choosing desired property filter
## Bug description

This is happening across insights and /events and can somewhat be captured at the end of [this recording](https://app.posthog.com/recordings?filters=%7B%22actions%22%3A%5B%5D%2C%22events%22%3A%5B%5D%2C%22properties%22%3A%5B%7B%22key%22%3A%22email%22%2C%22value%22%3A%5B%22cameron%40posthog.com%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22person%22%7D%5D%2C%22date_from%22%3A%222022-08-23%22%2C%22date_to%22%3Anull%2C%22offset%22%3A0%2C%22session_recording_duration%22%3A%7B%22type%22%3A%22recording%22%2C%22key%22%3A%22duration%22%2C%22value%22%3A60%2C%22operator%22%3A%22gt%22%7D%7D#sessionRecordingId=18338b7ef492e1-0888b53b905cbb-1a525635-13c680-18338b7ef4a2314) - despite obscured filter, I'm trying to select 'contains' instead of 'equals' but as soon as I do, the filter applies before anything can be selected and it runs the query. 

## How to reproduce

1. go to /events and select a person property (email) and then switch the filter type from `equals` to `contains` and the query will run before you can enter anything
2.
3.

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Closes https://github.com/PostHog/posthog/issues/7970
Closes https://github.com/PostHog/posthog/issues/11814

If a Popup has content that opens another popup, it isn't technically "outside" but from a DOM perspective it is meaning propagated events can't be detected.

This has always been an issue for the TaxonomicFilter but is made worse with the addition of move to more Lemon components which use the Popup internally rather than an inline popup which is what Antd (mostly but not always) used.

## Changes

It's not pretty but:
1. Popups listen for insideClicks
2. If there is a parent popup (checking context) then set a global flag (yuck) to say it was handled (if the additional popup property is not set i.e. `closeParentPopupOnClickInside=false`
3. The parent popup which receives the "outsideClick" checks if this flag is set on the next tick. If it is, it ignores it
4. The nested popup on the next tick (which is after the parent) picks this up and sets the flag back to false

### Nested menu popup works as before

Works by setting the flag `closeParentPopupOnClickInside=true`. This has to be done for all nestedt LemonButtonWithPopups for menus. I actually think we should create a more config-driven `LemonMenu` which would allow us to control all the popups together.

### Previously broken `PropertyFilters` now work.

As all other popups have the default `closeParentPopupOnClickInside=false`, clicks inside the LemonSelect popup for example are recorded and then the parent's outsideClickHandler will be prevented from running
 

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by benjackwhite** [comment]
A bit more testing of this and it doesnt work as desired. Particularly with menus. The sub-popups of something like Export struggle in this solution as technically they are a nested popup so the click is ignored, leaving the core menu open...

**Comment by liyiy** [comment]
We probably want to be moving towards a more unified property filters design too, most of the updates now lay the filters out (like how insights does it) so that it doesn't disappear and we can technically avoid this issue. Might just need a design eye on how to lay these filters out when they're on pages like `events` and `persons` but I think that's a good direction to move in?

**Comment by mariusandra** [comment]
I once tried tracking nested popups through nested contexts... but quickly gave up. I don't remember the specifics, but there were for sure many different types of popups (e.g. antd dropdowns, `Popup`s, etc) and it felt tedious. 

I guess I'm trying to say good luck 😅

**Review by pauldambra — APPROVED** [review]
clicking around and it seems to work

It's a bit surprising when it does close. If the filter considers that the selection is complete then it closes. But I don't think that's new here.

**Comment by pauldambra** [comment]
I would love if this had a cypress test. 

when you click X then filter opens
when you click Y it stays open
when you click Z it closes

@benjackwhite

File: frontend/src/lib/components/Popup/Popup.tsx [hunk 1]
```diff
@@ -144,7 +152,15 @@ export const Popup = React.forwardRef<HTMLDivElement, PopupProps>(
...
             [floatingRef, referenceRef, ...additionalRefs],
-            (event) => visible && onClickOutside?.(event),
+            (event) => {
+                // Delay by a tick to allow other Popups to detect inside clicks.
+                // If a nested popup has handled the click, don't do anything
+                setTimeout(() => {
+                    if (visible && !nestedPopupReceivedClick) {
+                        onClickOutside?.(event)
+                    }
+                }, 1)
```
Comment (mariusandra) [inline frontend/src/lib/components/Popup/Popup.tsx:162]: Technically `setTimeout(() => {}, 1)` might delay by more than a tick if the ticks are really fast.

**Pull request merged**